src/gui/raster/qgscolorrampshaderwidget.cpp

```diff
@@ -645,9 +644,20 @@ void QgsColorRampShaderWidget::mColormapTreeWidget_itemEdited( QTreeWidgetItem *
 
 void QgsColorRampShaderWidget::setFromShader( const QgsColorRampShader &colorRampShader )
 {
+  populateColormapTreeWidget( colorRampShader.colorRampItemList() );
+
+  // Those objects are connected to classify() the color ramp shader if they change, or call widget change
+  // need to block them to avoid to classify and to alter the color ramp, or to call duplicate widget change
+  whileBlocking( mClipCheckBox )->setChecked( colorRampShader.clip() );
+  whileBlocking( mColorInterpolationComboBox )->setCurrentIndex( mColorInterpolationComboBox->findData( colorRampShader.colorRampType() ) );
+  mColorInterpolationComboBox_currentIndexChanged( mColorInterpolationComboBox->currentIndex() );
+  whileBlocking( mClassificationModeComboBox )->setCurrentIndex( mClassificationModeComboBox->findData( colorRampShader.classificationMode() ) );
+  mClassificationModeComboBox_currentIndexChanged( mClassificationModeComboBox->currentIndex() );
```

Comment:
why you block  the signal currentIndexChanged on line above and immediately emits it? isn't is the same as removal of `whileBlocking` line above?

Reply:
Blocking to avoid to call the slots `QgsColorRampShaderWidget::classify` that is also connected.